packages/host/app/components/matrix/forgot-password.gts

```diff
@@ -253,14 +263,11 @@ export default class ForgotPassword extends Component<Signature> {
   constructor(owner: Owner, args: any) {
     super(owner, args);
 
-    if (
-      this.matrixService.authState.sid &&
-      this.matrixService.authState.clientSecret
-    ) {
+    if (this.args.resetPasswordParams) {
       this.state = {
         type: 'resetPassword',
-        sid: this.matrixService.authState.sid,
-        clientSecret: this.matrixService.authState.clientSecret,
+        sid: this.args.resetPasswordParams.sid,
```

Comment:
Can you just read `sid` and `clientSecret` directly from `this.args.resetPasswordParams`? That way you can have a single source of truth rather than creating new state